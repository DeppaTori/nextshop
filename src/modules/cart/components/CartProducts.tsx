import {
  TableContainer,
  Paper,
  Table,
  TableHead,
  TableRow,
  TableCell,
  TableBody,
  Button,
} from "@mui/material";
import { MAP_TEST_ID } from "../../../constants";
import { CartProduct } from "../../../entity/CartProduct";
import { IDRFormatCurrency } from "../../../helpers/mixin";

interface CartProductProps {
  products: CartProduct[];
  increaseOnClick: Function;
  decreaseOnClick: Function;
}

export const CartProducts = ({
  products,
  increaseOnClick,
  decreaseOnClick,
}: CartProductProps) => {
  return (
    <TableContainer component={Paper}>
      <Table>
        <TableHead>
          <TableRow>
            <TableCell>No.</TableCell>
            <TableCell>Item Name</TableCell>
            <TableCell align="center">Quantity</TableCell>
            <TableCell align="right">Price</TableCell>
            <TableCell align="right">Total Price</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {products.map((product, i) => (
            <TableRow key={i}>
              <TableCell>{i + 1}</TableCell>
              <TableCell>{product.productName}</TableCell>
              <TableCell align="center">
                <Button
                  data-testid={MAP_TEST_ID.CART_INCREASE_QUANTITY_BTN}
                  onClick={() =>
                    increaseOnClick(product.productId, product.quantity)
                  }
                >
                  +
                </Button>
                <input
                  placeholder="quantity"
                  onChange={() => {}}
                  value={product.quantity}
                />
                <Button
                  data-testid={MAP_TEST_ID.CART_DECREASE_QUANTITY_BTN}
                  onClick={() =>
                    decreaseOnClick(product.productId, product.quantity)
                  }
                >
                  -
                </Button>
              </TableCell>
              <TableCell align="right">
                {IDRFormatCurrency(product.productPrice)}
              </TableCell>
              <TableCell align="right">
                {IDRFormatCurrency(product.totalPrice)}
              </TableCell>
            </TableRow>
          ))}
          <TableRow>
            <TableCell colSpan={4} align="right">
              Total
            </TableCell>
            <TableCell align="right">
              {IDRFormatCurrency(products[0].totalPrice)}
            </TableCell>
          </TableRow>
        </TableBody>
      </Table>
    </TableContainer>
  );
};
