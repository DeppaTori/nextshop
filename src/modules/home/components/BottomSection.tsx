export const BottomSection = () => {
  return (
    <div>
      <p>NextShop @2022</p>
    </div>
  );
};
