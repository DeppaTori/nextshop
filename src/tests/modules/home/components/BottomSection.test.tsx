import { render } from "@testing-library/react";
import { BottomSection } from "../../../../modules/home/components/BottomSection";

describe("BottomSection", () => {
  it("renders year", () => {
    const { container } = render(<BottomSection />);
    expect(container).toMatchSnapshot();
  });
});
