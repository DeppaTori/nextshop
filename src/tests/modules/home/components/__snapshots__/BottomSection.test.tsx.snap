// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomSection renders year 1`] = `
<div>
  <div>
    <p>
      NextShop @2022
    </p>
  </div>
</div>
`;
